File: JavaScript_Courses/JavaScript_Basics/Preparation/03_Conditional_Statements_Advanced_Exercise/03_New_House.js
function newHouse(input) {
    let rosePrice = 5;
    let dahliasPrice = 3.80;
    let tulipPrice = 2.80;
    let narcissusPrice = 3;
    let gladiolusPrice = 2.50;

    let flowersType = input[0];
    let flowersCount = Number(input[1]);
    let budget = Number(input[2]);

    let totalCost = 0;
    if (flowersType == "Roses") {
        totalCost = flowersCount * rosePrice;
        if (flowersCount > 80) {
            totalCost *= 0.90;
        }
    } else if (flowersType == "Dahlias") {
        totalCost = flowersCount * dahliasPrice;
        if (flowersCount > 90) {
            totalCost *= 0.85;
        }
    } else if (flowersType == "Tulips") {
        totalCost = flowersCount * tulipPrice;
        if (flowersCount > 80) {
            totalCost *= 0.85;
        }
    } else if (flowersType == "Narcissus") {
        totalCost = flowersCount * narcissusPrice;
        if (flowersCount < 120) {
            totalCost *= 1.15;
        }
    } else if (flowersType == "Gladiolus") {
        totalCost = flowersCount * gladiolusPrice;
        if (flowersCount < 80) {
            totalCost *= 1.20;
        }
    }

    if (budget >= totalCost) {
        console.log(`Hey, you have a great garden with ${flowersCount} ${flowersType} and ${(budget - totalCost).toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money, you need ${(totalCost - budget).toFixed(2)} leva more.`);
    }
}

// newHouse(["Roses", "55", "250"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/03_Conditional_Statements_Advanced_Exercise/03_New_House.py
flowers_type = input()  # "Roses", "Dahlias", "Tulips", "Narcissus", "Gladiolus"
flowers_count = int(input())
budget = int(input())

flower_price = 0.0
discount = 0.0
if flowers_type == 'Roses':
    flower_price = 5.00
    if flowers_count > 80:
        discount = 0.10
elif flowers_type == 'Dahlias':
    flower_price = 3.80
    if flowers_count > 90:
        discount = 0.15
elif flowers_type == 'Tulips':
    flower_price = 2.80
    if flowers_count > 80:
        discount = 0.15
elif flowers_type == 'Narcissus':
    flower_price = 3.00
    if flowers_count < 120:
        discount = -0.15
elif flowers_type == 'Gladiolus':
    flower_price = 2.50
    if flowers_count < 80:
        discount = -0.20

total_price = flower_price * flowers_count * (1 - discount)

if budget >= total_price:
    money_left = budget - total_price
    print(f'Hey, you have a great garden with {flowers_count} {flowers_type} and {money_left :.2f} leva left.')
else:
    money_needed = total_price - budget
    print(f'Not enough money, you need {money_needed :.2f} leva more.')
